gpflow/base.py

```diff
@@ -107,41 +146,41 @@ def log_prior_density(self):
             return log_p
 
     @property
-    def prior_on(self):
+    def prior_on(self) -> Union[str, PriorOn]:
```

Comment:
```suggestion
    def prior_on(self) -> PriorOn:
```

Reply:
good point. as an aside to this PR, are users going to get confused if they do `param.prior_on = "constrained"` and then find that `param.prior_on != "constrained"`?

Reply:
I think that's an acceptable risk compared to the value of only ever having to compare against the actual enum in our code. There's already a bunch of implicit conversions from numpy to tensor...